### DISCUSSION TITLE
Stored Procedures or .sql file execution with paramter/variable support

### DISCUSSION THREAD

**Discussion by Jasonmitche11** [duckdb/duckdb#13092] [Ideas]
Hello All,

Very new to duckdb, but extremely impressed so far!!!

Just curious to see if duckdb stand alone installation currently supports one of the following abilities: 
1) create a stored procedure accepting parameter and defining variables
2) execute a series of queries from a .sql file

Problem:
I have the same set of queries, I need to run on a recurring basis to derive a final data set. The files are time relative so it would be nice to pass in the date parameters to be referenced in the query filters when the queries are executed. Looking for a way to either execute the same set of queries passing in variables to be used in query statements or at a minimum executing the SQL statements from a file such as passing the file with SQL statements to parser.

Any support, guidance or insights would be much appreciated.

Best Regards

**Comment by heejit** [comment]
Something like pl/pgsql or pl/python will be really helpfull

**Comment by Pete1999** [comment]
I was hoping to use DuckDb instead of sqlite. I currently use the abstraction of Postgresql sql and pl/pgsql functions to do all my database work. Want to port from Postgresql to a lightweight db for app development. The lack of this feature is a deal breaker, might as well use sqlite. Not the main use case for DuckDb, but a very nice use for it, in my opinion. Along with schemas, this would make it a very good alternate to sqlite.

Revision: I investigated using a macro in the way I am now using Postrgresql sql functions. Not useful since macros can not be used for multiple sql statements.
